src/components/inputs/Slider/Handle.tsx

```diff
@@ -37,7 +38,10 @@ export function Handle({
     <HandleStyled
       focused={focused === handle || dragging === handle}
       style={{ left: (value / max) * 100 + '%' }}
-      onMouseDown={() => !disabled && setDragging(handle)}
+      onMouseDown={() => {
+        !disabled && setDragging(handle);
+      }}
+      onMouseUp={() => dragging && onDragEnd()}
```

Comment:
Hey @sbatel! A warning would get passed here if onDragEnd does not get passed. So...a couple things:

- `mouseup` is already being tracked in [this useEffect](https://github.com/vimeo/iris/blob/main/src/components/inputs/Slider/Slider.tsx#L72-L78).  I would omit the `dragging` check here.
- Instead, a similar check as `onMouseDown` could be done, as well as whether `onDragEnd` has been passed, since it's an optional prop and React would throw a warning if it wasn't passed
`!disabled && onDragEnd && onDragEnd()`

Reply:
Hi! Thanks! i missed that useEffect. It makes much more sense to use it there.
so i removed the onMouseUp that i added and used the function there (after checking if exists)

Reply:
Apologies for the misunderstanding, you were going in the right direction with putting `onDragEnd()` on the `onMouseUp`. If it's called on the `useEffect`, the component will _always_ be listening to whether you're dragging _anywhere_ on the window, even if it's not on the slider. 

I meant that this should be the check passed to `onMouseUp`:
```
onMouseUp ={() => !disabled && onDragEnd && onDragEnd()}
```